src/merkle/smt/full/mod.rs

```diff
@@ -323,6 +319,382 @@ impl Smt {
     }
 }
 
+// Concurrent implementation
+#[cfg(feature = "concurrent")]
+impl Smt {
+    /// Parallel implementation of [`Smt::with_entries()`].
+    ///
+    /// This method constructs a new sparse Merkle tree concurrently by processing subtrees in
+    /// parallel, working from the bottom up. The process works as follows:
+    ///
+    /// 1. First, the input key-value pairs are sorted and grouped into subtrees based on their leaf
+    ///    indices. Each subtree covers a range of 256 (2^8) possible leaf positions.
+    ///
+    /// 2. The subtrees are then processed in parallel:
+    ///    - For each subtree, compute the inner nodes from depth D down to depth D-8.
+    ///    - Each subtree computation yields a new subtree root and its associated inner nodes.
+    ///
+    /// 3. These subtree roots are recursively merged to become the "leaves" for the next iteration,
+    ///    which processes the next 8 levels up. This continues until the final root of the tree is
+    ///    computed at depth 0.
+    pub fn with_entries_concurrent(
```

Comment:
Does this need to be `pub` or could it be `pub(crate)`, for example?

Same questions about `with_entries_sequential()`.